test/index.js

```diff
@@ -123,208 +123,163 @@ describe('Schwifty', () => {
-
-                const plugin1 = (srv, opts, next) => {
-
-                    srv.schwifty(getOptions({
-                        models: [
-                            TestModels.Dog,
-                            TestModels.Person
-                        ]
-                    }));
-
-                    // Monkeypatch the destroy func
-                    const oldDestroy = srv.knex().destroy;
-                    srv.knex().destroy = (cb) => {
-
-                        ++toredown;
-                        return oldDestroy(cb);
-                    };
+        const server = await getServer(getOptions());
 
-                    next();
-                };
+        let toredown = 0;
 
-                plugin1.attributes = { name: 'plugin-one' };
+        await server.initialize();
+        expect(toredown).to.equal(0);
 
-                const plugin2 = (srv, opts, next) => {
+        const plugin1 = {
+            name: 'plugin-one',
+            register: (srv, opts) => {
 
-                    srv.schwifty([TestModels.Zombie]);
+                // Creates plugin-specific knex instance using the base connection configuration specified in getOptions
+                srv.schwifty(getOptions({
+                    models: [
+                        TestModels.Dog,
+                        TestModels.Person
+                    ]
+                }));
 
-                    // Plugin 2 will use server.root's knex connection
-                    expect(srv.knex()).to.shallow.equal(srv.root.knex());
+                // Monkeypatch the destroy func
+                const oldDestroy = srv.knex().destroy;
+                srv.knex().destroy = () => {
 
-                    next();
+                    ++toredown;
+                    // Returns a Promise, which is await'd in lib/index::internals.stop
+                    return oldDestroy();
                 };
 
-                plugin2.attributes = { name: 'plugin-two' };
+            }
+        };
 
-                const oldDestroy = server.knex().destroy;
-                server.knex().destroy = (cb) => {
+        const plugin2 = {
+            name: 'plugin-two',
+            register: (srv, opts) => {
 
-                    ++toredown;
-                    return oldDestroy(cb);
-                };
-
-                server.register([plugin1, plugin2], (err) => {
+                srv.schwifty([TestModels.Zombie]);
 
-                    expect(err).to.not.exist();
+                // Plugin 2 will use server.root's knex connection
+                // Referencing server.knex() is a bit of a hacky though required workaround to to inspect the root server's knex() decoration, given that hapi17 removed server.root ( this test previously used srv.root.knex() )
+                // In this case, because we can be certain that server is the root server for plugin2, we can also be certain that this comparison will work. There is no guarantee such referencing will work in
```

Comment:
Are you concerned about cases where you just don't happen to have a reference to the root server?  That is totally fine!

Reply:
oh, cool, thanks for taking a peek at this.

ay, yup, that was my concern!  

but cool, glad that's all fine. why is that fine? guessing something to do with there being very few cases where you need a reference to your root server beyond contrived / test cases like this?

just generally still trying to wrap my head around the realm stuff completely 🙏

Reply:
I think the best way to think of this is how it's meant to be used in practice.  We wouldn't expect for users to reach to the root server explicitly using `server.root` (especially now that it's not part of hapi).  But users still might have access to the root server organically (e.g. because they explicitly created a hapi server and aren't in a plugin).  Whether it's implemented with realms or something else, the user doesn't care– this is an integration-style test, which has a "bird's eye" view on the entire setup and is meant to test behavior of schwifty rather than its implementation.

Reply:
ahhh gotcha. yea, that distinction between "bird's eye" vs. ..."user's eye"(?) is super helpful. cool cool, I'll remove the comment. thanks for the lesson, my dude! 🙏